### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by talevy**

### CODE REVIEW

File: lib/logstash/inputs/dlq.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+require 'logstash/namespace'
+require 'logstash/inputs/base'
+require 'logstash-input-dlq.jar'
+
+
+class LogStash::Inputs::Dlq < LogStash::Inputs::Base
+  config_name 'dlq'
```
Comment (suyograo) [inline lib/logstash/inputs/dlq.rb]: I think we need to be consistent. Elsewhere, we've used the full dead_letter_queue (in core config) and here its DLQ. dead_letter_queue is long, but it's self-explanatory to users who are not familiar with this terminology.

File: lib/logstash/inputs/dlq.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  default :codec, 'plain'
+
+  config :path, :validate => :path, :required => true
+  config :sincedb, :validate => :string, :required => false
```
Comment (suyograo) [inline lib/logstash/inputs/dlq.rb]: By default this should be in `path.data/plugins/inputs/dead_letter_queue/`. That's what file input uses. Also rename to sincedb_path to match file input.

File: lib/logstash/inputs/dlq.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  config :commit_offsets, :validate => :boolean, :default => true
+  config :timestamp, :validate => :string, :required => false
```
Comment (suyograo) [inline lib/logstash/inputs/dlq.rb]: Maybe call it `from_timestamp` or `start_timestamp`. Just a `timestamp` may not be very obvious.

**Review by suyograo — CHANGES_REQUESTED** [review]

**Comment by talevy** [comment]
Merging to Close the loop on this. Work still needs to be done before publishing. This is a good start for iterative improvements until version 1.0

**Pull request merged**